cdk-infra/lib/constructs/auto-builder-api-construct.ts

```diff
@@ -0,0 +1,151 @@
+    });
+
+    const fastlyDochubToken = StringParameter.fromSecureStringParameterAttributes(this, 'fastlyDochubToken', {
+      parameterName: '/env/dev/docs/worker_pool/fastly/docs/dochub/token',
+    });
+    const fastlyDochubServiceId = StringParameter.fromSecureStringParameterAttributes(this, 'fastlyDochubServiceId', {
+      parameterName: '/env/dev/docs/worker_pool/fastly/docs/dochub/service_id',
+    });
+    const fastlyDochubMap = StringParameter.fromSecureStringParameterAttributes(this, 'fastlyDochubMap', {
+      parameterName: '/env/dev/docs/worker_pool/fastly/dochub_map',
+    });
+
+    const dochubTriggerUpsertLambda = new NodejsFunction(this, 'dochubTriggerUpsertLambda', {
+      entry: `${HANDLERS_PATH}/dochub.ts`,
+      runtime,
+      handler: 'UpsertEdgeDictionaryItem',
+      environment,
+    });
+
+    fastlyDochubToken.grantRead(dochubTriggerUpsertLambda);
+    fastlyDochubServiceId.grantRead(dochubTriggerUpsertLambda);
+    fastlyDochubMap.grantRead(dochubTriggerUpsertLambda);
+
+    const githubTriggerLambda = new NodejsFunction(this, 'githubTriggerLambda', {
+      entry: `${HANDLERS_PATH}/github.ts`,
+      runtime,
+      handler: 'TriggerBuild',
+      bundling,
+      environment,
+    });
+
+    const triggerLocalBuildLambda = new NodejsFunction(this, 'triggerLocalBuildLambda', {
+      entry: `${HANDLERS_PATH}/jobs.ts`,
+      runtime,
+      handler: 'TriggerLocalBuild',
+      environment,
+      bundling,
+    });
+
+    // generic handler for the root endpoint
+    const rootEndpointLambda = new Function(this, 'RootEndpointLambda', {
+      code: Code.fromInline('exports.default = (event) => { console.log("hello, world!!"); }'),
+      runtime,
+      handler: 'RootEndpointLambda',
+    });
+
+    const restApi = new LambdaRestApi(this, 'webhookHandlers', {
+      handler: rootEndpointLambda,
+      proxy: false,
+    });
+
+    const webhookEndpoint = restApi.root.addResource('webhook');
+
+    const slackEndpoint = webhookEndpoint.addResource('slack');
+    const dochubEndpoint = webhookEndpoint.addResource('dochub');
+    const githubEndpoint = webhookEndpoint.addResource('githubEndpoint');
+    const localEndpoint = webhookEndpoint.addResource('local');
+
+    const defaultCorsPreflightOptions: CorsOptions = {
+      allowOrigins: Cors.ALL_ORIGINS,
+    };
+
+    // add resources and post methods for trigger endpoints
+    slackEndpoint
+      .addResource('trigger')
+      .addResource('build', { defaultCorsPreflightOptions })
+      .addMethod('POST', new LambdaIntegration(slackTriggerLambda));
+
+    slackEndpoint
+      .addResource('display')
+      .addResource('repos', { defaultCorsPreflightOptions })
+      .addMethod('POST', new LambdaIntegration(slackDisplayRepoLambda));
+
+    dochubEndpoint
+      .addResource('trigger')
+      .addResource('upsert', { defaultCorsPreflightOptions })
+      .addMethod('POST', new LambdaIntegration(dochubTriggerUpsertLambda));
+
+    githubEndpoint
```

Comment:
Will building this endpoint through AWS CDK simply replace the current Lambda deployments for each existing env with whatever is in the v2 file? Or will the new endpoints constructed here be on a different hostname?

Asking to check if this will require any of our users to swap out their webhooks

Reply:
Good question, it will create a new URL for these APIs.